pkg/iptables/iptables.go

```diff
@@ -344,11 +290,31 @@ func deleteRuleRaw(iptablesCmd string, table Table, rule ...string) (string, err
 	}
 
 	// Delete rule
-	cmd = exec.Command(iptablesCmd, append(append(args, "-D"), rule...)...)
-	out, err = cmd.CombinedOutput()
+	out, err = runIptables(iptablesCmd, table, append([]string{"-D"}, rule...)...)
 	if err != nil {
 		return string(out), err
 	}
 
 	return string(out), nil
 }
+
+// Run iptables with lock
+func runIptables(iptablesCmd string, table Table, args ...string) ([]byte, error) {
+	// Lock
+	lock.Lock()
```

Comment:
Lock must be in each function, not here. This lock is for race conditions between function in iptables package, not between iptables command.

Reply:
I put the locks back where they were.